lib/cfndsl/module.rb

```diff
@@ -65,12 +66,22 @@ def dsl_content_object(*symbols)
               hash = {}
               instance_variable_set(pluralvar, hash)
             end
-            hash[name] ||= definition_class.new(*values)
-            hash[name].instance_eval(&block) if block
-            return hash[name]
+            instance = hash[name]
```

Comment:
For direct usage we get more scenarios where the objects are re-used, so we need some more protection from doing things like redeclaring objects with different classes, or trying to reinitialise an object that already exists.